File: Programming Basics/JavaScript/While loops/Labs/account-balance.js
function solve(input) {
    let amount = 0;

    let index = 0;
    while (input[index] != "NoMoreMoney") {
        const operation = Number(input[index]);
        if (operation < 0) {
            console.log("Invalid operation!");
            break;
        }

        console.log(`Increase: ${operation.toFixed(2)}`);
        amount += operation;
        index++;
    }

    console.log(`Total: ${amount.toFixed(2)}`);
}

File: Programming Basics/C++/While loops/Labs/AccountBalance/AccountBalance.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    cout.setf(ios::fixed);
    cout.precision(2);

    double amount = 0;

    string input;
    while (cin >> input && input != "NoMoreMoney")
    {
        double operation = stod(input);

        if (operation < 0)
        {
            cout << "Invalid operation!" << endl;
            break;
        }

        cout << "Increase: " << operation << endl;
        amount += operation;
    }

    cout << "Total: " << amount << endl;
}
